app/discovery/health.go

```diff
@@ -0,0 +1,99 @@
+package discovery
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+)
+
+// CheckResult is result of health-check
```

Comment:
maybe making it HealtChecker and injecting to discovery a better aprouch here? This way we separate concerns better and could just use interfaces in both consumers

Reply:
I don't think I got it right. Shouldn't it be the other way around?
Health check changes internal state of service mappers (`dead`). In my latest commit I implemented `HealthCheck` in  `discovery` package. Then I added method `CheckHealth` to the `Matcher` interface in `proxy` package.  Please let me know if you meant differently.

Reply:
with the current implementation, I don't have a strong opinion anymore but what I meant is to enclose the functionality of your current CheckHealth() method into a struct and define HealthChecker interface with a single method Check.